<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgMaterialsManager</class>
 <widget class="QDialog" name="DlgMaterialsManager">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>420</width>
    <height>374</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Select Material</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Material ID:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QComboBox" name="comboBoxMaterial">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Select a Material from the Catalog</string>
        </property>
        <property name="whatsThis">
         <string>Select a Material from the Catalog to view/edit exposure timing</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Description:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QLineEdit" name="lineEditDescription">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="palette">
         <palette>
          <active>
           <colorrole role="Text">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>127</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>204</red>
              <green>204</green>
              <blue>204</blue>
             </color>
            </brush>
           </colorrole>
          </active>
          <inactive>
           <colorrole role="Text">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>127</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>204</red>
              <green>204</green>
              <blue>204</blue>
             </color>
            </brush>
           </colorrole>
          </inactive>
          <disabled>
           <colorrole role="Text">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>120</red>
              <green>120</green>
              <blue>120</blue>
             </color>
            </brush>
           </colorrole>
           <colorrole role="Base">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>240</red>
              <green>240</green>
              <blue>240</blue>
             </color>
            </brush>
           </colorrole>
          </disabled>
         </palette>
        </property>
        <property name="toolTip">
         <string>Description of the selected Material</string>
        </property>
        <property name="whatsThis">
         <string>Description of the selected Material</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pushButtonAdd">
        <property name="toolTip">
         <string>Add a new Material to the Catalog</string>
        </property>
        <property name="whatsThis">
         <string>Add a new Material to the Catalog</string>
        </property>
        <property name="text">
         <string>Add...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButtonDuplicate">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Add a duplicate of the selected material to the Catalog.</string>
        </property>
        <property name="whatsThis">
         <string>Add a duplicate of the selected material to the Catalog.  This is good if you wish to modify an existing material.</string>
        </property>
        <property name="text">
         <string>Duplicate...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="pushButtonDelete">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Remove the selected material from the Catalog.  Default entries can not be deleted.</string>
        </property>
        <property name="whatsThis">
         <string>Remove the selected material from the Catalog.  Default entries can not be deleted.</string>
        </property>
        <property name="text">
         <string>Delete Selected</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QDialogButtonBox" name="buttonBoxSaveCancel">
     <property name="toolTip">
      <string>Click Save All to save your changes or Cancel to discard them.</string>
     </property>
     <property name="whatsThis">
      <string>Click Save All to save your changes or Cancel to discard them.</string>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::SaveAll</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_ExposureSettings">
     <property name="title">
      <string>Material Eposure Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Selected XY Pixel Resolution Table</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBoxXY">
        <property name="toolTip">
         <string>Select the xy resolution for data to display below.</string>
        </property>
        <property name="whatsThis">
         <string>Select the xy resolution for data to display below.  The printer can be calibrated to these xy pixel sizes. </string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>15</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>First (attach) Layer(s) Cure Time(sec):</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox">
        <property name="toolTip">
         <string>Time to cure the intial (attach) layer.</string>
        </property>
        <property name="whatsThis">
         <string>Time to cure the intial (attach) layer.  This time is independent of the layer thickness as we always start at zero.  This is to provide a solid reference for the rest of the layers.</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.500000000000000</double>
        </property>
        <property name="maximum">
         <double>60.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.500000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Number of First (attach) Layers:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="spinBoxNumberOfAttachLayers">
        <property name="toolTip">
         <string>Number of Layers to cure for the above length of time</string>
        </property>
        <property name="whatsThis">
         <string>Number of Layers to cure for the above length of time</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>For the selected XY Pixel Resolution Table, enter the base exposure time and the additional over cure exposure time for each layer.  Enter Zero if printing at this layer thickness is not allowed.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QTableWidget" name="tableWidget">
        <property name="toolTip">
         <string>Exposure settings for a given material and xy pixel size.</string>
        </property>
        <property name="whatsThis">
         <string>Exposure settings for a given material and xy pixel size.  Each possible layer thickness and it's associated timing is shown.</string>
        </property>
        <column>
         <property name="text">
          <string>T-Base (sec)</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>T-Over (sec)</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>comboBoxMaterial</tabstop>
  <tabstop>lineEditDescription</tabstop>
  <tabstop>pushButtonAdd</tabstop>
  <tabstop>pushButtonDuplicate</tabstop>
  <tabstop>pushButtonDelete</tabstop>
  <tabstop>comboBoxXY</tabstop>
  <tabstop>doubleSpinBox</tabstop>
  <tabstop>spinBoxNumberOfAttachLayers</tabstop>
  <tabstop>tableWidget</tabstop>
  <tabstop>buttonBoxSaveCancel</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBoxSaveCancel</sender>
   <signal>accepted()</signal>
   <receiver>DlgMaterialsManager</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBoxSaveCancel</sender>
   <signal>rejected()</signal>
   <receiver>DlgMaterialsManager</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
